packages/react-hooks/src/react-hooks.tsx

```diff
@@ -28,17 +28,8 @@ export async function preloadShape(
 }
 
 export function sortedOptionsHash(options: ShapeStreamOptions): string {
-  const shapeDef = JSON.stringify(
-    options.shape,
-    Object.keys(options.shape).sort()
-  )
-  // eslint-disable-next-line
-  const { shape, ...optionsWithoutShapeDef } = options
-  const allOptions = JSON.stringify(
-    optionsWithoutShapeDef,
-    Object.keys(options).sort()
-  )
-  const shapeHash = shapeDef + allOptions
+  const allOptions = JSON.stringify(options, Object.keys(options).sort())
+  const shapeHash = allOptions
```

Comment:
This assignment is not needed, could just return `allOptions`.

Reply:
The trick with providing the keys to be stringified and sorting that array of keys is neat although i'm not sure if it is safe to rely on this trick. I quickly looked at the docs and - unless i missed it - it does not explicitly state that the order of the properties in the stringified output will respect the order of the keys in the reviver argument.

Reply:
I was basing it off this stackoverflow answer — https://stackoverflow.com/a/16168003 — it seems pretty likely to me that the stringified output is written in order? It passes the test anyways... unless we have a better approach or you're sure it doesn't work, it seems fine.

Will fix `allOptions`